Added missing HTML filter for quantity

tomaskavalek opened the pull request:
New filter into `wc_add_to_cart_message` which allows modify printed product quantity into message – for example in case of floats units – 1.2 kg (`add_filter('woocommerce_stock_amount', 'floatval');`)

### All Submissions:

* [x] Have you followed the [WooCommerce Contributing guideline](https://github.com/woocommerce/woocommerce/blob/master/.github/CONTRIBUTING.md)?
* [x] Does your code follow the [WordPress' coding standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/)?
* [x] Have you checked to ensure there aren't other open [Pull Requests](../../pulls) for the same update/change?

<!-- Mark completed items with an [x] -->

<!-- You can erase any parts of this template not applicable to your Pull Request. -->

### Changes proposed in this Pull Request:

Creates a new filter to customize the shown number of quantity.

Globally it will be very good to create filter, or filters, for all `absint` to allow `float` values, with HTML modifier or good usage of `number_format_i18n`.

<!-- Describe the changes made to this Pull Request, and the reason for such changes. -->

Closes # .

### How to test the changes in this Pull Request:

1. Create a filter that uses the new hook, and returns `return $qty`.
2. Modify filter for `woocommerce_stock_amount` from `intval` to `floatval`.

### Other information:

* [x] Have you added an explanation of what your changes do and why you'd like us to include them?
* [ ] Have you written new tests for your changes, as applicable?
* [x] Have you successfully ran tests with your changes locally?

<!-- Mark completed items with an [x] -->

### Changelog entry

> Enter a short summary of all changes on this Pull Request. This will appear in the changelog if accepted.

tomaskavalek committed:
Added missing HTML filter for quantity

New filter into wc_add_to_cart_message which allows modify printed product quantity into message – for example in case of floats units – 1.2 kg (add_filter('woocommerce_stock_amount', 'floatval');)

claudiulodro reviewed (dismissed):
Hi there,

This sounds like a reasonable filter to add.

This change has a couple issues:
1. The product id is what you are currently filtering and what will get output if you use the `woocommerce_add_to_cart_message_qty_html` filter.
2.  This line of code has become very confusing and hard to read.

My recommendation is to wrap the entire quantity html in a filter, then it is easier to read and more flexible. Something like this:

```
$titles[] = apply_filters( 'woocommerce_add_to_cart_qty_html', ( $qty > 1 ? absint( $qty ) . ' &times; ' : '' ), $product_id ) . apply_filters( 'woocommerce_add_to_cart_item_name_in_quotes', sprintf( _x( '&ldquo;%s&rdquo;', 'Item name in quotes', 'woocommerce' ), strip_tags( get_the_title( $product_id ) ) ), $product_id );
```

kloon commented:
@tomaskavalek just checking if you saw @claudiulodro's request? If you can add those changes we can get this merged.

tomaskavalek committed:
Added missing HTML filter for quantity

https://github.com/woocommerce/woocommerce/pull/21069#pullrequestreview-147005349

tomaskavalek commented:
@kloon Thank you a lot for notify. I did it here https://github.com/woocommerce/woocommerce/pull/21069/commits/f93f36ee65b972330c4357f79c8da5fd871f359a.
